packages/common/src/utils/filters.ts

```diff
@@ -520,12 +524,53 @@ export const addFiltersToMetricQuery = (
     },
 });
 
+const findAndOverrideChartFilter = (
+    item: FilterGroupItem,
+    filterRulesList: FilterRule[],
+): FilterGroupItem => {
+    const overridableDashboardFilter = isFilterRule(item)
```

Comment:
Note that there might be multiple dashboard filters for the same field. 

We should add unit tests for the following cases:

- if a dashboard has multiple filters for the same field, we replace them as a nested filter group.

```
// chart filters 
status=updated OR is_complete=true

// dashboard filters 
status=updated AND status not null

// result
(status=updated AND status not null) OR is_complete=true
```

- if the chart has multiple filters for the same field, we replace all of them

```
// chart filters 
status=updated OR status is null

// dashboard filters 
status=updated

// result
status=updated OR status=updated
```